src/ContinuosBondingERC20Token.sol

```diff
@@ -201,45 +209,69 @@ contract ContinuosBondingERC20Token is ERC20, ReentrancyGuard {
         currentEth = currentEth > address(this).balance ? address(this).balance : currentEth;
         isLpCreated = true;
 
-        //is this line necessary? can be removed
-        // _approve(address(this), address(router), currentTokenBalance);
-
-        address wNative;
-        uint256 liquidity;
-        if (poolType == LP_POOL.Uniswap) {
-            wNative = IUniswapV2Router02(router).WETH();
-
-            address pair = factory.getPair(address(this), wNative);
-            if (pair == address(0)) pair = factory.createPair(address(this), wNative);
-            IWETH(wNative).deposit{ value: currentEth }();
-            IWETH(wNative).transfer(pair, currentEth);
-            _transfer(address(this), pair, currentTokenBalance);
-            liquidity = IPair(pair).mint(address(this));
-        } else if (poolType == LP_POOL.TraderJoe) {
-            wNative = IJoeRouter02(router).WAVAX();
-
-            address pair = factory.getPair(address(this), wNative);
-            if (pair == address(0)) pair = factory.createPair(address(this), wNative);
-            IWETH(wNative).deposit{ value: currentEth }();
-            IWETH(wNative).transfer(pair, currentEth);
-            _transfer(address(this), pair, currentTokenBalance);
-            liquidity = IPair(pair).mint(address(this));
-        }
-        // Burn the LP tokens received
-        IERC20 lpToken = IERC20(factory.getPair(address(this), wNative));
-        lpToken.transfer(BURN_ADDRESS, liquidity);
-
-        emit PairCreated(currentEth, currentTokenBalance, liquidity, address(lpToken));
+        // Create the pool if it doesn't exist
+        (address token0, address token1) = address(this) < WETH ? (address(this), WETH) : (WETH, address(this));
+        (uint256 amountOfToken0, uint256 amountOfToken1) =
+            (token0 == address(this)) ? (currentTokenBalance, currentEth) : (currentEth, currentTokenBalance);
+        address pool = nonfungiblePositionManager.createAndInitializePoolIfNecessary(
+            token0, token1, 3000, _getSqrtPriceX96(amountOfToken0, amountOfToken1)
+        );
+
+        // Approve the position manager to spend tokens
+        _approve(address(this), address(nonfungiblePositionManager), currentTokenBalance);
+        IWETH(WETH).deposit{ value: currentEth }();
+        IWETH(WETH).approve(address(nonfungiblePositionManager), currentEth);
+
+        // Add liquidity
+        INonfungiblePositionManager.MintParams memory params = INonfungiblePositionManager.MintParams({
+            token0: address(this),
+            token1: WETH,
+            fee: 3000,
+            tickLower: TickMath.MIN_TICK,
+            tickUpper: TickMath.MAX_TICK,
+            amount0Desired: currentTokenBalance,
+            amount1Desired: currentEth,
+            amount0Min: 0,
+            amount1Min: 0,
+            recipient: IBondingERC20TokenFactory(factory).feeRecipient() != address(0)
```

Comment:
would this spend more gas by read the feeRecipient twice?

Reply:
did the change